src/H5Iint.c

```diff
@@ -9663,20 +9376,18 @@ H5I__discard_mt_type_info(H5I_mt_type_info_t * type_info_ptr)
     result = atomic_compare_exchange_strong(&(type_info_ptr->on_fl), &on_fl, TRUE);
     assert(result);
 
+    atomic_store(&(type_info_ptr->serial_num), atomic_fetch_add(&(H5I_mt_g.type_next_sn), 1ULL));
```

Comment:
Same as the analysis for ID info, I don't think this guarantees a unique SN for each entry